### PR TITLE
Add deploy workflow

### LINKED ISSUE #2: Setup publication workflow
Such that the documentation id published in GitHub Pages when there's a new push to the main branch. See, for example: https://github.com/ImperialCollegeLondon/FINESSE/blob/f4bda58b7ff03facd8d2b9e0f8468acad67ff402/.github/workflows/ci.yml#L96

### PR DESCRIPTION
**Pull request by alexdewar**
I've done this using the new GitHub Actions approach.

Closes #2.

### CODE REVIEW

**Comment by alexdewar** [comment]
I wasn't able to test that this was working, but I *think* it's correct (famous last words).

Previously I've been able to test docs workflows like this by playing around with the repo settings, but I can't find the options. Maybe I'm not an admin? Dunno.

**Review by dalonsoa — APPROVED** [review]
There more things modified than required as `pre-commit` has been enabled and has modified a few things, but otherwise it looks good to me. A lot of options, like the concurrency issue, I did not know that existed!

I don't know why you cannot run this manually. You're all maintainers and should have permission to do that sort of things. I'll check again.

File: mkdocs.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-site_name: Reseach Software Engineering Team Blog
+site_name: Research Software Engineering Team Blog
```
Comment (dc2917) [inline mkdocs.yml:1]: Was this a codespell fix? Just curious as to whether it's actually any good 😅

**Review by dc2917 — APPROVED** [review]
Looks good to me!

File: mkdocs.yml (hunk 1 above)
Comment (dalonsoa) [reply mkdocs.yml:1]: That one in particular I had to do myself, but all the others were automatic.

**Comment by alexdewar** [comment]
Cool. Thanks for fixing things up @dalonsoa!

**Comment by alexdewar** [comment]
Deployed successfully. Huzzah!

**Pull request merged**